test/test-spawn.c

```diff
@@ -1367,6 +1367,79 @@ TEST_IMPL(spawn_setgid_fails) {
 }
 #endif
 
+TEST_IMPL(spawn_affinity) {
+#if defined(_WIN32) || defined(__linux__)
+  int i;
+  int r;
+  int cpu;
+  char cpustr[11];
+  char* newmask;
+  int cpumask_size;
+#ifdef _WIN32
+  DWORD_PTR procmask;
+  DWORD_PTR sysmask;
+#else
+  cpu_set_t cpuset;
+#endif
+
+  cpumask_size = uv_cpumask_size();
+  ASSERT(cpumask_size > 0);
+
+  /* find a cpu we can use */
+  cpu = cpumask_size;
+#ifdef _WIN32
+  r = GetProcessAffinityMask(GetCurrentProcess(), &procmask, &sysmask);
+  ASSERT(r != 0);
+  for (i = 0; i < cpumask_size; ++i) {
+    if (procmask & (((DWORD_PTR)1) << i)) {
+      cpu = i;
+      break;
+    }
+  }
+#else
+  CPU_ZERO(&cpuset);
+  r = pthread_getaffinity_np(pthread_self(), sizeof(cpuset), &cpuset);
+  ASSERT(r == 0);
+  for (i = 0; i < cpumask_size; ++i) {
+    if (CPU_ISSET(i, &cpuset)) {
+      cpu = i;
+      break;
+    }
+  }
+#endif
+  ASSERT(cpu < cpumask_size);
+  snprintf(cpustr, sizeof(cpustr), "%d", cpu);
+
+  init_process_options("spawn_helper_affinity", exit_cb);
+
+  /* mask the child to just one cpu */
+  newmask = (char*)calloc(cpumask_size, 1);
+  ASSERT(newmask != NULL);
+  newmask[cpu] = 1;
+  options.cpumask_size = (size_t)cpumask_size;
+  options.cpumask = newmask;
+
+  /* tell the child which one it should get */
+  options.args[2] = cpustr;
+  options.args[3] = "dummy"; /* need 4 args for test/run-tests.c dispatch */
+
```

Comment:
I woul add a test where cpumask_size were less than required to check it fails with `UV_EINVAL`

Reply:
Good idea.  Added as another test case.